### PR TITLE
[Review] Remove orm coupling and register recalculator dynamically. 

### PR DESCRIPTION
**Pull request by lchrusciel**
| Q               | A
| --------------- | ---
| Bug fix?        | yes |
| New feature?    | no |
| BC breaks?      | yes |
| Related tickets | fixes #7250  |
| License         | MIT |

### CODE REVIEW

File: src/Sylius/Bundle/ReviewBundle/DependencyInjection/SyliusReviewExtension.php [hunk 1]
```diff
@@ -57,15 +63,33 @@ private function resolveResources(array $resources, ContainerBuilder $container)
...
      * @param ContainerBuilder $container
      */
-    private function addProperTagToReviewDeleteListener(ContainerBuilder $container)
+    private function createReviewListeners(array $reviewSubjects, ContainerBuilder $container)
     {
-        if (!$container->hasDefinition('sylius.listener.review_delete')) {
-            return;
-        }
+        foreach ($reviewSubjects as $reviewSubject) {
+            $reviewRatingUpdater =  new Definition(AverageRatingUpdater::class, [
```
Comment (pamil) [inline src/Sylius/Bundle/ReviewBundle/DependencyInjection/SyliusReviewExtension.php]: Doubled space! 🐋

File: src/Sylius/Bundle/ReviewBundle/DependencyInjection/SyliusReviewExtension.php [hunk 2]
```diff
@@ -57,15 +63,33 @@ private function resolveResources(array $resources, ContainerBuilder $container)
...
+                    new Reference('sylius.average_rating_calculator'),
+                    new Reference(sprintf('sylius.manager.%s_review', $reviewSubject)),
+            ]);
+            $reviewChangeListener =  new Definition(ReviewChangeListener::class, [
```
Comment (pamil) [inline src/Sylius/Bundle/ReviewBundle/DependencyInjection/SyliusReviewExtension.php]: Doubled space! 🐋

File: src/Sylius/Bundle/ReviewBundle/DependencyInjection/SyliusReviewExtension.php [hunk 3]
```diff
@@ -57,15 +63,33 @@ private function resolveResources(array $resources, ContainerBuilder $container)
...
      */
-    private function addProperTagToReviewDeleteListener(ContainerBuilder $container)
+    private function createReviewListeners(array $reviewSubjects, ContainerBuilder $container)
     {
-        if (!$container->hasDefinition('sylius.listener.review_delete')) {
-            return;
-        }
+        foreach ($reviewSubjects as $reviewSubject) {
+            $reviewRatingUpdater =  new Definition(AverageRatingUpdater::class, [
+                    new Reference('sylius.average_rating_calculator'),
```
Comment (pamil) [inline src/Sylius/Bundle/ReviewBundle/DependencyInjection/SyliusReviewExtension.php]: Too deep indentation level 🌮

File: src/Sylius/Bundle/ReviewBundle/DependencyInjection/SyliusReviewExtension.php [hunk 4]
```diff
@@ -57,15 +63,33 @@ private function resolveResources(array $resources, ContainerBuilder $container)
...
+                    new Reference(sprintf('sylius.manager.%s_review', $reviewSubject)),
+            ]);
+            $reviewChangeListener =  new Definition(ReviewChangeListener::class, [
+                    new Reference(sprintf('sylius.%s_review.average_rating_updater', $reviewSubject)),
```
Comment (pamil) [inline src/Sylius/Bundle/ReviewBundle/DependencyInjection/SyliusReviewExtension.php]: Too deep indentation level 🌮

**Comment by pjedrzejewski** [comment]
Thank you Łukasz!

**Comment by videni** [comment]
@lchrusciel, thank you

**Comment by lchrusciel** [comment]
@videni you're welcome :)

**Pull request merged**